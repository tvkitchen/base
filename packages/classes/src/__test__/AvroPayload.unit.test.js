import fs from 'fs'
import path from 'path'
import { AvroPayload } from '..'

describe('AvroPayload', () => {
	describe('serialize', () => {
		it('should properly should serialize data with an empty object', () => {
			jest.clearAllMocks()
			const mockDate = new Date(1592156234000)
			jest.spyOn(global, 'Date').mockImplementation(() => mockDate)
			const payload = new AvroPayload()
			jest.spyOn(global, 'Date').mockRestore()
			const serializedAvroPayload = AvroPayload.serialize(payload)
			expect(serializedAvroPayload).toMatchSnapshot()
		})
		it('should properly should serialize data with a partially populated object', () => {
			const parameters = {
				data: 'I ate all of the cheese',
				createdAt: '2020-07-08T17:28:29.032Z',
			}
			const payload = new AvroPayload(parameters)
			const serializedAvroPayload = AvroPayload.serialize(payload)
			expect(serializedAvroPayload).toMatchSnapshot()
		})
		it('should properly should serialize data with a fully populated object', () => {
			const parameters = {
				data: 'I ate all of the cheese',
				type: 'CONFESSION',
				createdAt: '2020-02-02T03:04:05.000Z',
				timestamp: '2020-02-02T03:04:01.000Z',
				duration: 1000,
				position: 60000,
			}
			const payload = new AvroPayload(parameters)
			const serializedAvroPayload = AvroPayload.serialize(payload)
			expect(serializedAvroPayload).toMatchSnapshot()
		})
	})

	describe('deserialize', () => {
		it('should properly should deserialize an empty serialized payload', () => {
			const payload = AvroPayload.deserialize('{}')
			expect(payload.data).toBeNull()
			expect(payload.type).toBeNull()
		})
		it('should properly deserialize a partially populated serialized payload', () => {
			const payload = AvroPayload.deserialize('{"data":"Hello."}')
			expect(payload.data).toBe('Hello.')
			expect(payload.type).toBeNull()
		})
		it('should properly deserialize a serialized payload', () => {
			const parameters = {
				data: 'I ate all of the cheese',
				type: 'CONFESSION',
				createdAt: '2020-02-02T03:04:05.000Z',
				timestamp: '2020-02-02T03:04:01.000Z',
				duration: 1000,
				position: 60000,
			}
			const payload = new AvroPayload(parameters)
			const serializedAvroPayload = AvroPayload.serialize(payload)
			const deserializedAvroPayload = AvroPayload.deserialize(serializedAvroPayload)
			expect(deserializedAvroPayload).toMatchSnapshot()
		})
		it('should properly deserialize payload strings with null values', () => {
			const serializedAvroPayload = '{"data":"hello","type":null}'
			const deserializedAvroPayload = AvroPayload.deserialize(serializedAvroPayload)
			expect(deserializedAvroPayload).toMatchSnapshot({
				createdAt: expect.any(String),
			})
		})
		it('should properly deserialize a serialized payload containing buffered data', () => {
			const data = fs.readFileSync(path.join(__dirname, '/data/thinkingface.png'))
			const parameters = {
				data,
				type: 'CONFESSION',
			}
			const payload = new AvroPayload(parameters)
			const serializedAvroPayload = AvroPayload.serialize(payload)
			const deserializedAvroPayload = AvroPayload.deserialize(serializedAvroPayload)
			expect(deserializedAvroPayload).toEqual(payload)
		})
	})
})
