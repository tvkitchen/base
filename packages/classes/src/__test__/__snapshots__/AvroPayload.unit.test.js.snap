// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvroPayload deserialize should properly deserialize a serialized payload 1`] = `
AvroPayload {
  "createdAt": "2020-02-02T03:04:05.000Z",
  "data": "I ate all of the cheese",
  "duration": 1000,
  "origin": "2020-02-02T03:04:01.000Z",
  "position": 60000,
  "type": "CONFESSION",
}
`;

exports[`AvroPayload deserialize should properly deserialize payload strings with null values 1`] = `
Object {
  "createdAt": Any<String>,
  "data": "hello",
  "duration": 0,
  "origin": "",
  "position": 0,
  "type": null,
}
`;

exports[`AvroPayload serialize should properly should serialize data with a fully populated object 1`] = `
Object {
  "data": Array [
    4,
    46,
    73,
    32,
    97,
    116,
    101,
    32,
    97,
    108,
    108,
    32,
    111,
    102,
    32,
    116,
    104,
    101,
    32,
    99,
    104,
    101,
    101,
    115,
    101,
    2,
    20,
    67,
    79,
    78,
    70,
    69,
    83,
    83,
    73,
    79,
    78,
    2,
    48,
    50,
    48,
    50,
    48,
    45,
    48,
    50,
    45,
    48,
    50,
    84,
    48,
    51,
    58,
    48,
    52,
    58,
    48,
    53,
    46,
    48,
    48,
    48,
    90,
    2,
    48,
    50,
    48,
    50,
    48,
    45,
    48,
    50,
    45,
    48,
    50,
    84,
    48,
    51,
    58,
    48,
    52,
    58,
    48,
    49,
    46,
    48,
    48,
    48,
    90,
    208,
    15,
    192,
    169,
    7,
  ],
  "type": "Buffer",
}
`;

exports[`AvroPayload serialize should properly should serialize data with a partially populated object 1`] = `
Object {
  "data": Array [
    4,
    46,
    73,
    32,
    97,
    116,
    101,
    32,
    97,
    108,
    108,
    32,
    111,
    102,
    32,
    116,
    104,
    101,
    32,
    99,
    104,
    101,
    101,
    115,
    101,
    0,
    2,
    48,
    50,
    48,
    50,
    48,
    45,
    48,
    55,
    45,
    48,
    56,
    84,
    49,
    55,
    58,
    50,
    56,
    58,
    50,
    57,
    46,
    48,
    51,
    50,
    90,
    2,
    0,
    0,
    0,
  ],
  "type": "Buffer",
}
`;

exports[`AvroPayload serialize should properly should serialize data with an empty object 1`] = `
Object {
  "data": Array [
    0,
    0,
    2,
    48,
    50,
    48,
    50,
    48,
    45,
    48,
    54,
    45,
    49,
    52,
    84,
    49,
    55,
    58,
    51,
    55,
    58,
    49,
    52,
    46,
    48,
    48,
    48,
    90,
    2,
    0,
    0,
    0,
  ],
  "type": "Buffer",
}
`;
